packages/client/src/profile/profileReducer.ts

```diff
@@ -143,19 +144,31 @@ export const profileReducer: LoopReducer<
     case actions.MODIFY_USER_DETAILS:
       const details: IUserDetails = action.payload
 
+      const commandList: (
+        | RunCmd<Action>
+        | ActionCmd<changeLanguageActions.Action>
+      )[] = [
+        Cmd.run(storeUserDetails, {
+          args: [details]
+        })
+      ]
+      if (state.userDetails?.language !== details.language) {
+        commandList.push(
+          Cmd.action(
+            changeLanguageActions.changeLanguage({
+              language: details.language
+            })
+          )
+        )
+      }
+
       if (details) {
         return loop(
           {
-            ...state
+            ...state,
+            userDetails: details
           },
-          Cmd.list([
-            Cmd.run(() => storeUserDetails(details)),
-            Cmd.action(
-              changeLanguageActions.changeLanguage({
-                language: details.language
-              })
-            )
-          ])
+          Cmd.list(commandList)
```

Comment:
Previously whenever this action was called it always called `changeLanguage` action in loop. Now it should only call that if the language actually changed. Unfortunately, I wasn't able to test this as we only have a single language at the moment.